src/frontend/src/util.js

```diff
@@ -7,6 +7,161 @@ export function parseTaxaID(taxaString) {
+      const interLat = p1[0] + t * (p2[0] - p1[0]);
+      // refer to linear interpolation formula for a better understanding
+
+      // push two points, one for west, one for east hemisphere
+      intersections.push({
+        p: p1,
+        pIdx: i,
+        lat: interLat,
+        lng: 180 * Math.sign(p1[1]),
+      });
+
+      intersections.push({
+        p: p2,
+        pIdx: (i + 1) % boundary.length,
+        lat: interLat,
+        lng: 180 * Math.sign(p2[1]),
+      });
+    }
+  }
+
+  return intersections;
+}
+
+/**
+ * Checks if a boundary crosses the dateline, lng=+-180
+ *
+ * @param {Array<Array<number>>} boundary - List of [latitude, longitude] coordinates
+ * @returns {boolean} - Returns true if the boundary crosses the dateline, false otherwise
+ */
+export function crossesDateLine(boundary) {
+  // checks if a polygon
+  let crossesDateline = false;
+  for (let i = 0; i < boundary.length; i++) {
+    const p1 = boundary[i];
+    const p2 = boundary[(i + 1) % boundary.length];
+
+    // if longer distance is greater than 180, the shortest path crosses dateline
+    if (Math.abs(p1[1] - p2[1]) > 180) {
+      crossesDateline = true;
+      break;
+    }
+  }
+  return crossesDateline; // boolean
+}
+
+/**
+ * Checks if a boundary crosses the Date Line and splits the polygon into two parts if it does
+ *
+ * @param {Array<Array<number>>} boundary - List of [latitude, longitude] coordinates
+ * @returns {Array<Array<Array<number>>>} - Array of two polygons or a single polygon, based on if it crosses dateline
+ */
+export function checkAndSplitBoundary(boundary) {
+  // checks if hexagon crosses dateline, if so splits to render it properly
+
+  // must sort boundary to apply check and split operations
+  const sortedBoundary = sortBoundaryClockwise(boundary);
+  // will not split boundary if it does not cross dateline
+  if (!crossesDateLine(sortedBoundary)) {
+    return [sortedBoundary];
+  }
+
+  const intersections = findDateLineIntersectionsOnBoundary(sortedBoundary);
+  const leftPoly = [];
+  const rightPoly = [];
+
+  for (let i = 0; i < sortedBoundary.length; i++) {
+    const p = sortedBoundary[i];
+
+    // depending on point's lng, push it to left or right
+    // also push the intersection point related to the point
+    if (p[1] < 0) {
+      leftPoly.push(p);
+      const inter = intersections.find((int) => int.pIdx === i);
+      if (inter) {
+        leftPoly.push([inter.lat, inter.lng]);
+      }
+    } else {
+      rightPoly.push(p);
+      const inter = intersections.find((int) => int.pIdx === i);
+      if (inter) {
+        rightPoly.push([inter.lat, inter.lng]);
+      }
+    }
+  }
+
+  // sort polygons again to avoid polygon collapse
+  const sortedLeft = sortBoundaryClockwise(leftPoly);
+  const sortedRight = sortBoundaryClockwise(rightPoly);
+
+  // list gets extended by the caller
+  return [sortedLeft, sortedRight];
+}
+
```

Comment:
These functions look good to me! Good job figuring out the trickiness around intersections and sorting the polygon points for display. This looks great as is, so no need to change anything now, but I wanted to give a couple words of advice for future work: 
- In the future you might consider using a spatial geometry library like https://turfjs.org/docs/api for some of these  functions. For example, https://turfjs.org/docs/api/lineIntersect would be a good alternative to the findDateLineIntersectionsOnBoundary and crossesDateLine functions you wrote. So long as you trust the library you use, this can help reduce chance of bugs and the amount of code you have to write. 
- These functions are good candidates for unit tests in the future, since they have a fair bit of complexity.